clients/src/main/java/org/apache/kafka/common/record/Compressor.java

```diff
@@ -201,16 +201,16 @@ public void put(final byte[] bytes, final int offset, final int len) {
         }
     }
 
-    public void putRecord(long timestamp, byte[] key, byte[] value, CompressionType type, int valueOffset, int valueSize) {
+    public void putRecord(long timestamp, long crc, byte[] key, byte[] value, CompressionType type,
+                          int valueOffset, int valueSize) {
         // put a record as un-compressed into the underlying stream
-        long crc = Record.computeChecksum(timestamp, key, value, type, valueOffset, valueSize);
         byte attributes = Record.computeAttributes(type);
         putRecord(crc, attributes, timestamp, key, value, valueOffset, valueSize);
 
     }
 
-    public void putRecord(long timestamp, byte[] key, byte[] value) {
-        putRecord(timestamp, key, value, CompressionType.NONE, 0, -1);
+    public void putRecord(long timestamp, long crc, byte[] key, byte[] value) {
+        putRecord(timestamp, crc, key, value, CompressionType.NONE, 0, -1);
     }
 
     private void putRecord(final long crc, final byte attributes, final long timestamp, final byte[] key, final byte[] value, final int valueOffset, final int valueSize) {
```

Comment:
I n this method, it's `crc`, `attributes` and then `timestamp`. It would be good to be consistent when it comes to ordering the parameters.